src/transactions/OfferExchange.cpp

```diff
@@ -50,7 +50,7 @@ canSellAtMost(LedgerTxnHeader const& header, LedgerTxnEntry const& account,
         return std::max({getAvailableBalance(header, account), int64_t(0)});
     }
 
-    if (trustLine && trustLine.isAuthorized())
+    if (trustLine && trustLine.isAuthorizedToMaintainLiabilities())
```

Comment:
Do we actually need to check authorization here at all starting from version 10? Same thing for the other overload.

Reply:
This is a good point. We don't need to check as the authorization is checked prior to this for v10 and up. I did this because it looked cleaner than `header.current().ledgerVersion >= 10 || trustLine.isAuthorized()`, but I think the intention is clearer if we only check the authorization pre v10. I'll make the change.